src/Avalonia.Visuals/Media/FontManager.cs

```diff
@@ -53,60 +129,13 @@ public string DefaultFontFamilyName
         /// <returns>
         ///     The matched typeface.
         /// </returns>
-        public abstract Typeface MatchCharacter(int codepoint, FontWeight fontWeight = default,
-            FontStyle fontStyle = default,
-            FontFamily fontFamily = null, CultureInfo culture = null);
-
-        public static FontManager CreateDefault()
-        {
-            var platformImpl = AvaloniaLocator.Current.GetService<IFontManagerImpl>();
-
-            if (platformImpl != null)
-            {
-                return new PlatformFontManager(platformImpl);
-            }
-
-            return new EmptyFontManager();
-        }
-
-        private class PlatformFontManager : FontManager
-        {
-            private readonly IFontManagerImpl _platformImpl;
-
-            public PlatformFontManager(IFontManagerImpl platformImpl)
-            {
-                _platformImpl = platformImpl;
-
-                DefaultFontFamilyName = _platformImpl.DefaultFontFamilyName;
-            }
-
-            public override IEnumerable<string> GetInstalledFontFamilyNames(bool checkForUpdates = false) =>
-                _platformImpl.GetInstalledFontFamilyNames(checkForUpdates);
-
-            public override Typeface GetCachedTypeface(FontFamily fontFamily, FontWeight fontWeight, FontStyle fontStyle) =>
-                _platformImpl.GetTypeface(fontFamily, fontWeight, fontStyle);
-
-            public override Typeface MatchCharacter(int codepoint, FontWeight fontWeight = default,
-                FontStyle fontStyle = default,
-                FontFamily fontFamily = null, CultureInfo culture = null) =>
-                _platformImpl.MatchCharacter(codepoint, fontWeight, fontStyle, fontFamily, culture);
-        }
-
-        private class EmptyFontManager : FontManager
+        public Typeface MatchCharacter(int codepoint, FontWeight fontWeight = FontWeight.Normal,
```

Comment:
Ah interesting, so this is how one does e.g. a fallback to Chinese or certain emojis when the specified font doesn't support that character? I notice this method isn't used currently, will it be called in subsequent PRs?

Reply:
It will be called by the new text layout or some custom code. This is also culture aware. So the found font will always be the best fit. This also runs threw all FontFamilys that are defined in the FontFamily object.